src/Swashbuckle.AspNetCore.Annotations/SwaggerParameterAttribute.cs

```diff
@@ -30,5 +32,37 @@ public bool Required
         }
 
         internal bool? RequiredFlag { get; set; }
+
+        /// <summary>
+        /// Describes how the parameter value will be serialized depending on the type of
+        /// the parameter value. Default values (based on value of in): for query - form;
+        /// for path - simple; for header - simple; for cookie - form.
+        /// </summary>
+        public string Style
+        {
+            get { throw new InvalidOperationException($"Use {nameof(ParameterStyle)} instead"); }
+            set
+            {
+                ParameterStyle = Enum.TryParse(value, ignoreCase: true, out ParameterStyle result) ? result :
+                    throw new InvalidOperationException(
+                        message: $"Style '{value}' not defined in OpenAPI specification");
+            }
+        }
+
+        internal ParameterStyle? ParameterStyle { get; set; }
+
+        /// <summary>
+        /// When this is true, parameter values of type array or object generate separate parameters for
+        /// each value of the array or key-value pair of the map. For other types of parameters this property
+        /// has no effect. When style is form, the default value is true. For all other styles,
+        /// the default value is false.
+        /// </summary>
```

Comment:
I would simplify this explanation so we're not copying too many implementation details that could change in the future into our documentation.